src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/TextFieldDriver.cs

```diff
@@ -25,6 +40,50 @@ public override IDisplayResult Edit(TextField field, BuildFieldEditorContext con
         {
             return Initialize<EditTextFieldViewModel>(GetEditorShapeType(context), model =>
             {
+                var settings = context.PartFieldDefinition.Settings["TextFieldPredefinedListEditorSettings"]?.ToObject<TextFieldPredefinedListEditorSettings>();
```

Comment:
Why are you changing the TextField driver? An editor setting should be in its own driver.

Reply:
Yeah I think I should move this to TextFieldPredefinedListEditorSettingsDriver Edit method.